packages/gatsby/src/utils/jobs-manager.ts

```diff
@@ -250,29 +270,26 @@ exports.createInternalJob = (job, plugin) => {
 
 /**
  * Creates a job
- *
- * @param {InternalJob} job
- * @return {Promise<object>}
  */
-exports.enqueueJob = async job => {
+export async function enqueueJob(job: InternalJob): Promise<object> {
   // When we already have a job that's executing, return the same promise.
   // we have another check in our createJobV2 action to return jobs that have been done in a previous gatsby run
   if (jobsInProcess.has(job.contentDigest)) {
-    return jobsInProcess.get(job.contentDigest).deferred.promise
+    return jobsInProcess.get(job.contentDigest)!.deferred.promise
```

Comment:
Since jobs are deleted from the map, wouldn't it be _possible_ for this to be `undefined`?